### PR TITLE
Add java api for topic/partition API

### PR DESCRIPTION
**Pull request by 13h3r**
Java api for #139. I am not very familiar with java-scala interop neither java 8, so please advise if I miss something.

### CODE REVIEW

File: core/src/main/scala/akka/kafka/KafkaConsumerActor.scala [hunk 1]
```diff
@@ -21,28 +21,28 @@ object KafkaConsumerActor {
...
-    private[kafka] final case class Revoked(partition: List[TopicPartition])
-    private[kafka] final case class Messages[K, V](messages: Iterator[ConsumerRecord[K, V]])
-    private[kafka] final case class Committed(offsets: Map[TopicPartition, OffsetAndMetadata])
+    final case class Assigned(partition: List[TopicPartition])
+    final case class Revoked(partition: List[TopicPartition])
+    final case class Messages[K, V](messages: Iterator[ConsumerRecord[K, V]])
+    final case class Committed(offsets: Map[TopicPartition, OffsetAndMetadata])
     //internal
     private[KafkaConsumerActor] case object Poll
+    val number = new AtomicInteger()
```
Comment (ktoso) [inline core/src/main/scala/akka/kafka/KafkaConsumerActor.scala]: can this be private?

File: core/src/main/scala/akka/kafka/javadsl/Consumer.scala [hunk 2]
```diff
@@ -88,5 +93,49 @@ object Consumer {
...
+   * you have lot of manually assigned topic-partitions and want to keep only one kafka consumer
+   */
+  def plainExternalSource[K, V](consumer: ActorRef, subscription: ManualSubscription): Source[ConsumerRecord[K, V], Control] = {
+    scaladsl.Consumer.plainExternalSource(consumer, subscription)
+      .mapMaterializedValue(new WrappedConsumerControl(_))
+      .asJava
+  }
+
+  /**
+   * The same as `plainExternalSource` but with offset commit support
```
Comment (ktoso) [inline core/src/main/scala/akka/kafka/javadsl/Consumer.scala]: could be `[[plainExternalSource]]` for better linking?

File: core/src/main/scala/akka/kafka/javadsl/Consumer.scala [hunk 3]
```diff
@@ -88,5 +93,49 @@ object Consumer {
...
 
+  /**
+   * The `plainPartitionedSource` is a way to track automatic partition assignment from kafka.
+   * When topic-partition is assigned to a consumer this source will emit tuple with assigned topic-partition and a corresponding source
+   * When topic-partition is revoked then corresponding source completes
+   */
+  def plainPartitionedSource[K, V](settings: ConsumerSettings[K, V], subscription: AutoSubscription): Source[Pair[TopicPartition, Source[ConsumerRecord[K, V], NotUsed]], Control] = {
+    scaladsl.Consumer.plainPartitionedSource(settings, subscription)
+      .map {
+        case (tp, source) => Pair(tp, source.asJava)
```
Comment (ktoso) [inline core/src/main/scala/akka/kafka/javadsl/Consumer.scala]: `map { Pair(_, _) }` won't work?

File: core/src/main/scala/akka/kafka/KafkaConsumerActor.scala (hunk 1 above)
Comment (13h3r) [inline core/src/main/scala/akka/kafka/KafkaConsumerActor.scala]: yes

File: core/src/main/scala/akka/kafka/javadsl/Consumer.scala (hunk 2 above)
Comment (13h3r) [inline core/src/main/scala/akka/kafka/javadsl/Consumer.scala]: ok

File: core/src/main/scala/akka/kafka/javadsl/Consumer.scala (hunk 3 above)
Comment (13h3r) [inline core/src/main/scala/akka/kafka/javadsl/Consumer.scala]: No, `map { Pair(_, _) }` requires two arguments, but we have just one (`Tuple2`)

**Comment by patriknw** [comment]
Take a look at `Subscriptions` also. There are some scala `Set` and `Map` there that might need Java counterpart.

File: core/src/main/scala/akka/kafka/javadsl/Consumer.scala (hunk 3 above)
Comment (ktoso) [inline core/src/main/scala/akka/kafka/javadsl/Consumer.scala]: Technically this works: `map { Pair.tupled }`, nevermind though

Comment (13h3r) [inline core/src/main/scala/akka/kafka/javadsl/Consumer.scala]: Then we probably need to make `asJava` conversion implicit

**Comment by patriknw** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
@13h3r let me know when it's a good time to release a new milestone

**Pull request merged**